delfin/drivers/netapp/dataontap/netapp_handler.py

```diff
@@ -0,0 +1,469 @@
+            Tools.split_value_map(disk_str, disks_map, split=':')
+            logical_type = constant.DISK_LOGICAL. \
+                get(disks_map['ContainerType'])
+            """Map disk physical information"""
+            for physical_info in physicals_list:
+                if len(physical_info) > 6:
+                    if physical_info[0] == disks_map['k']:
+                        physical_type = \
+                            constant.DISK_TYPE.get(physical_info[1])
+                        speed = physical_info[5]
+                        firmware = physical_info[4]
+            status = constants.DiskStatus.ABNORMAL
+            if disks_map['Errors:'] is None or disks_map['Errors:'] == "":
+                status = constants.DiskStatus.NORMAL
+            disk_model = {
+                'name': disks_map['k'],
+                'storage_id': storage_id,
+                'native_disk_id': disks_map['k'],
+                'serial_number': disks_map['SerialNumber'],
+                'manufacturer': disks_map['Vendor'],
+                'model': disks_map['Model'],
+                'firmware': firmware,
+                'speed': speed,
+                'capacity':
+                    int(Tools.get_capacity_size(disks_map['PhysicalSize'])),
+                'status': status,
+                'physical_type': physical_type,
+                'logical_type': logical_type,
+                'health_score': '',
+                'native_disk_group_id': disks_map['Aggregate'],
+                'location': '',
+            }
+            disks_list.append(disk_model)
+        return disks_list
+
+    def get_filesystems(self, storage_id):
+        fs_list = []
+        fs_info = self.ssh_pool.do_exec(
+            constant.FS_SHOW_DETAIL_COMMAND)
+        fs_array = fs_info.split(
+            constant.FS_SPLIT_STR)
+        thin_fs_info = self.ssh_pool.do_exec(
+            constant.THIN_FS_SHOW_COMMAND)
+        pool_list = self.list_storage_pools(storage_id)
+        thin_fs_array = thin_fs_info.split("\r\n")
+        type = constants.FSType.THICK
+        fs_map = {}
+        for fs_str in fs_array[1:]:
+            Tools.split_value_map(fs_str, fs_map, split=':')
+            if fs_map is not None or fs_map != {}:
+                pool_id = ""
+                """get pool id"""
+                for pool in pool_list:
+                    if pool['name'] == fs_map['AggregateName']:
+                        pool_id = pool['native_storage_pool_id']
+                deduplicated = True
+                if fs_map['SpaceSavedbyDeduplication'] == '0B':
+                    deduplicated = False
+                if len(thin_fs_array) > 2:
+                    for thin_vol in thin_fs_array[2:]:
+                        thin_array = thin_vol.split()
+                        if len(thin_array) > 4:
+                            if thin_array[1] == fs_map['VolumeName']:
+                                type = constants.VolumeType.THIN
```

Comment:
`type` should have a default value in for loop, otherwise once the `type` was set to `constants.VolumeType.THIN`, it will always be `constants.VolumeType.THIN`

Reply:
Has been modified